File: springboot-azure/spring-boot-azure-sql-crud/src/main/java/com/knf/dev/demo/springbootazuresqlcrud/controller/UserController.java
package com.knf.dev.demo.springbootazuresqlcrud.controller;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.knf.dev.demo.springbootazuresqlcrud.entity.User;
import com.knf.dev.demo.springbootazuresqlcrud.exception.UserNotFound;
import com.knf.dev.demo.springbootazuresqlcrud.repository.UserRepository;

@RestController
@RequestMapping("/api/v1/users")
public class UserController {

	@Autowired
	UserRepository userRepository;

	// Create user
	@PostMapping
	public ResponseEntity<User> createUser(@RequestBody User user) {

		User newuser = new User(user.getFirstName(), user.getLastName(), user.getEmail());
		userRepository.save(newuser);
		return new ResponseEntity<>(newuser, HttpStatus.CREATED);

	}

	// Update user
	@PutMapping("/{id}")
	public ResponseEntity<User> updateUser(@PathVariable("id") Long id, @RequestBody User user) {

		Optional<User> userdata = userRepository.findById(id);
		if (userdata.isPresent()) {
			User _user = userdata.get();
			_user.setEmail(user.getEmail());
			_user.setFirstName(user.getFirstName());
			_user.setLastName(user.getLastName());
			return new ResponseEntity<>(userRepository.save(_user), HttpStatus.OK);
		} else {
			throw new UserNotFound("Invalid User Id");
		}
	}

	// Get all Users
	@GetMapping
	public ResponseEntity<List<User>> getAllUsers() {

		List<User> users = new ArrayList<User>();
		userRepository.findAll().forEach(users::add);
		return new ResponseEntity<>(users, HttpStatus.OK);

	}

	// Get user by ID
	@GetMapping("/{id}")
	public ResponseEntity<User> getUserByID(@PathVariable("id") Long id) {

		Optional<User> userdata = userRepository.findById(id);
		if (userdata.isPresent()) {
			return new ResponseEntity<>(userdata.get(), HttpStatus.OK);
		} else {
			throw new UserNotFound("Invalid User Id");
		}

	}

	// Delete user
	@DeleteMapping("/{id}")
	public ResponseEntity<User> deleteUser(@PathVariable("id") Long id) {

		Optional<User> userdata = userRepository.findById(id);
		if (userdata.isPresent()) {
			userRepository.deleteById(id);
			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
		} else {
			throw new UserNotFound("Invalid User Id");
		}
	}
}

File: kotlin-spring-main/kotlin_springb_h2db_thymeleaf_crud/src/main/kotlin/com/knf/dev/controller/UserController.kt
package com.knf.dev.controller

import com.knf.dev.entity.User
import com.knf.dev.repository.UserRepository
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.*
import org.springframework.ui.Model

import org.springframework.web.bind.annotation.GetMapping
import java.util.*
import org.springframework.web.bind.annotation.PathVariable


@Controller
class UserController(private val userRepository: UserRepository) {

    @GetMapping("/")
    fun getAllUsers(model: Model): String? {
        val list: List<User> = userRepository.findAll()
        model.addAttribute("users", list)
        return "list-users"
    }

    @PostMapping(path = ["/createUser"])
    fun createNewUser(@ModelAttribute user: User): String {
        userRepository.save(user)
        return "redirect:/";
    }

    @GetMapping(path = ["/add"])
    fun addUserById(): String? {
        return "add-user"
    }

    @GetMapping(path = ["/edit/{id}"])
    fun editUserById(model: Model, @PathVariable("id") id: Long): String? {
        model.addAttribute("user", userRepository.findById(id))
        return "edit-user"
    }

    @PostMapping(path = ["/editUser"])
    fun editUser(@ModelAttribute user: User): String? {
        userRepository.save(user)
        return "redirect:/"
    }

    @GetMapping(path = ["/delete/{id}"])
    fun deleteUserById(@PathVariable("id") id: Long): String? {
        userRepository.deleteById(id)
        return "redirect:/"
    }

}